src/lib.rs.in

```diff
@@ -173,7 +187,7 @@ impl Client {
     /// Take a nearly complete ElasticSearch URL, and stick the host/port part
     /// on the front.
     pub fn full_url(&self, suffix: &str) -> String {
-        format!("{}/{}", self.base_url, suffix)
+        format!("{}/{}", self.base_url.as_str(), suffix)
```

Comment:
Does `rust-url` (I haven't used it myself), support amending URLs?  If so it could be good to use that as a replacement of all the string concatenation already done.  It looks like there might be a `join` function that does just that: https://docs.rs/url/1.2.1/url/#base-url

But each of those return `Result` rather than a `String` so would have knock-on downstream effects.